packages/blocks/src/root-block/edgeless/components/frame/frame-preview.ts

```diff
@@ -78,8 +79,12 @@ export class FramePreview extends WithDisposable(ShadowlessElement) {
 
   private _frameDisposables: DisposableGroup | null = null;
 
-  private _getViewportWH = () => {
-    const [, , w, h] = deserializeXYWH(this.frame.xywh);
+  private _previewDoc: Doc | null = null;
+
+  private _previewSpec = SpecProvider.getInstance().getSpec('edgeless:preview');
+
+  private _updateFrameViewportWH = () => {
+    const [, , w, h] = deserializeXYWH(this.frame.xywh$.value);
```

Comment:
I think using `SignalWatcher` is better.

Reply:
I will remove this. Can SignalWatcher be used together with debounce? The intention here is to avoid re-rendering every time the frame xywh changes, as this could lead to performance issues. That's why we're listening to propUpdated and using debounce for updates.

Reply:
Indeed, it is true that it is no easy to implement debounce with signalwatcher. LGTM